elastic/elastic-agent
# Add `/liveness` endpoint to elastic-agent

## What does this PR do?

Closes https://github.com/elastic/elastic-agent/issues/390

## Why is it important?

This adds a `/liveness` endpoint to elastic agent that we enable along with the `processes` endpoint. However, unlike the `/processes` endpoint, `liveness` will return a `500` is any components are in an Unhealthy or Degraded state.

This is a pretty simple change as-is, since I figured it would be easier to discuss other theoretical changes when we can actually see code. In particular:

1) Should this be toggled by something other than the `monitoring.http` flag? Should we enabled it based on some k8s autodiscover config or env variable or something else? If this feature is fundamentally tied to a k8s workflow, it seems like enabling it should also be tied to k8s in some way.
2) Should we fail the health check based on any other conditions?

I'm also holding off on changing any docs until we're sure we have what we want with regards to config.

## How to test this PR

- set `monitoring.http.enabled` to `true`
- Check `/liveness/` and `/liveness/[component-id]` using either curl or a k8s health check.

## Checklist
